server/src/main/java/org/elasticsearch/search/builder/QueryCategory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.builder;
+
+public enum QueryCategory {
```

Comment:
Question: Would there be value in replacing `query_category` with a list of the actual query types that are entered? For example, you could imagine returning `query_types: [dense_vector, multi_match]`

This could potentially be more broadly useful than just for vector search, and robust to updating/adding new queries. The issue would be that in dashboards we'd need to know which fields we wanted to track for example, vector search. 

I'd love to hear your thoughts!

Reply:
Yes, we could do that. 👍 

My first thought was to be more coarse grained and try to categorize queries vs labeling them with all the query clauses. But we could go the other way round and start with the more fine grained approach and group them together in the dashboards themselves.

Reply:
What I like about that is we could take advantage of that monitoring for a lot of other search use cases!

Reply:
It would also be helpful for SDHs, so we could understand what components a query has.

My concern about that is, it's a bit over the initial scope. But I think it would be more beneficial in the mid term, plus we don't need to maintain the query categories.

Reply:
> My concern about that is, it's a bit over the initial scope. But I think it would be more beneficial in the mid term, plus we don't need to maintain the query categories.

Agreed that it's more beneficial in the long term and I think the initial time investment will pay off with less work later on.